sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1442 @@
+            case MIXED:
+                sentimentScores[0] = sentimentScore.getNegative();
+                sentimentScores[1] = sentimentScore.getNeutral();
+                sentimentScores[2] = sentimentScore.getPositive();
+                break;
+            default:
+                break;
+        }
+        return sentimentScores;
+    }
+
+    private TextSentimentClass convertToTextSentimentClass(final DocumentSentimentValue sentimentValue) {
+        switch (sentimentValue) {
+            case POSITIVE:
+                return TextSentimentClass.POSITIVE;
+            case NEUTRAL:
+                return TextSentimentClass.NEUTRAL;
+            case NEGATIVE:
+                return TextSentimentClass.NEGATIVE;
+            case MIXED:
+                return TextSentimentClass.MIXED;
+            default:
+                throw logger.logExceptionAsWarning(
+                    new RuntimeException(String.format("'%s' is not valid text sentiment.")));
+        }
+    }
+
+    private TextSentimentClass convertToTextSentimentClass(final SentenceSentimentValue sentimentValue) {
+        switch (sentimentValue) {
+            case POSITIVE:
+                return TextSentimentClass.POSITIVE;
+            case NEUTRAL:
+                return TextSentimentClass.NEUTRAL;
+            case NEGATIVE:
+                return TextSentimentClass.NEGATIVE;
+            default:
+                throw logger.logExceptionAsWarning(
+                    new RuntimeException(String.format("'%s' is not valid text sentiment.")));
+        }
+    }
+
+    private AnalyzeSentimentResult convertToErrorTextSentimentResult(final DocumentError documentError) {
+        final com.azure.ai.textanalytics.models.TextAnalyticsError error = convertToError(documentError.getError());
+        return new AnalyzeSentimentResult(documentError.getId(), null, error, null,
+            null);
+    }
+
+    /**
+     * Helper method to convert the service response of {@link LanguageResult} to {@link DocumentResultCollection}.
+     *
+     * @param languageResult the {@link LanguageResult} returned by the service.
+     *
+     * @return the {@link DocumentResultCollection} of {@link DetectLanguageResult} to be returned by the SDK.
+     */
+    private DocumentResultCollection<DetectLanguageResult> toDocumentResultCollection(
+        final LanguageResult languageResult) {
+        return new DocumentResultCollection<>(getDocumentLanguages(languageResult), languageResult.getModelVersion(),
+            languageResult.getStatistics() == null ? null : mapBatchStatistics(languageResult.getStatistics()));
+    }
+
+    /**
+     * Helper method to get a combined list of error documents and valid documents.
+     *
+     * @param languageResult the {@link LanguageResult} containing both the error and document list.
+     *
+     * @return the combined error and document list.
+     */
+    private List<DetectLanguageResult> getDocumentLanguages(final LanguageResult lang
```

Comment:
There are almost twenty methods in this class that are used only once (these convert methods or map). It adds additional complexity to this class and makes it hard to maintain.

Functions should be pieces of code that are reused throughout the class.

Reply:
Agree. We will refactor it to be more simple. But as we are going to give the private build to service team. I feel that we lack time on dealing with this refactor issue. Created an issue to track this concern after this week: https://github.com/Azure/azure-sdk-for-java/issues/6965

Reply:
Refactored methods to remove the ones only used once.